src/gt4py/next/program_processors/runners/dace_iterator/__init__.py

```diff
@@ -310,22 +322,31 @@ def run_dace_iterator(program: itir.FencilDefinition, *args, **kwargs):
 
     arg_types = [type_translation.from_value(arg) for arg in args]
 
-    cache_id = get_cache_id(program, arg_types, column_axis, offset_provider)
+    cache_id = get_cache_id(build_type, program, arg_types, column_axis, offset_provider)
     if build_cache is not None and cache_id in build_cache:
         # retrieve SDFG program from build cache
         sdfg_program = build_cache[cache_id]
         sdfg = sdfg_program.sdfg
 
     else:
-        sdfg = build_sdfg_from_itir(
-            program,
-            *args,
-            offset_provider=offset_provider,
-            auto_optimize=auto_optimize,
-            on_gpu=on_gpu,
-            column_axis=column_axis,
-            lift_mode=lift_mode,
-        )
+        # debug: test large icon4py stencils without regenerating the SDFG at each run
+        generate_sdfg = True
+        target = "gpu" if on_gpu else "cpu"
+        sdfg_filename = f"_dacegraphs/{target}/{program.id}.sdfg"
+
+        if generate_sdfg:
+            sdfg = build_sdfg_from_itir(
+                program,
+                *args,
+                offset_provider=offset_provider,
+                auto_optimize=auto_optimize,
+                on_gpu=on_gpu,
+                column_axis=column_axis,
+                lift_mode=lift_mode,
+            )
+            sdfg.save(sdfg_filename)
+        else:
+            sdfg = dace.SDFG.from_file(sdfg_filename)
```

Comment:
It looks like you have now two caches, one that is coming from the outside, `build_cache`  and one you create inside the `_dacegraphs` folder?

Reply:
Somehow, my comment got missing so I repeat it here:
I very much like the idea, but I think that it should be possible to influence the value from the outside, for example by an ENV or an explicit argument.
Because this is plain weird, you should at least put a comment to indicate that it must be changed manually.